src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2D.java

```diff
@@ -829,24 +829,27 @@ private void walkShape(Shape clip) throws IOException {
 		PathIterator pi = clip.getPathIterator(tf);
 		float[] coords = new float[6];
 		while (!pi.isDone()) {
-			switch (pi.currentSegment(coords)) {
-			case PathIterator.SEG_MOVETO:
-				contentStream.moveTo(coords[0], coords[1]);
-				break;
-			case PathIterator.SEG_LINETO:
-				contentStream.lineTo(coords[0], coords[1]);
-				break;
-			case PathIterator.SEG_QUADTO:
-				contentStream.curveTo1(coords[0], coords[1], coords[2], coords[3]);
-				break;
-			case PathIterator.SEG_CUBICTO:
-				contentStream.curveTo(coords[0], coords[1], coords[2], coords[3], coords[4], coords[5]);
-				break;
-			case PathIterator.SEG_CLOSE:
-				contentStream.closePath();
-				break;
+			int segment = pi.currentSegment(coords);
+			if(Float.isFinite(coords[0])) {
+				switch (segment) {
+				case PathIterator.SEG_MOVETO:
+					contentStream.moveTo(coords[0], coords[1]);
+					break;
+				case PathIterator.SEG_LINETO:
+					contentStream.lineTo(coords[0], coords[1]);
+					break;
+				case PathIterator.SEG_QUADTO:
+					contentStream.curveTo1(coords[0], coords[1], coords[2], coords[3]);
+					break;
+				case PathIterator.SEG_CUBICTO:
+					contentStream.curveTo(coords[0], coords[1], coords[2], coords[3], coords[4], coords[5]);
+					break;
+				case PathIterator.SEG_CLOSE:
+					contentStream.closePath();
+					break;
+				}
+				pi.next();
```

Comment:
You don't advanced the PathIterator if there is a invinite coords[0]? I Think that would cause a endless loop...